HildonLiveSearch improvement

le0nklcpp opened the pull request:
HAM HildonLiveSearch at this moment can only find requested package if a word in a package name/description begins with search string.
This patch allows to search for any occurrence in package name/description

le0nklcpp committed:
improve hildon livesearch

freemangordon commented:
the function you're modifying is called 'live_search_look_for_prefix' , after the change the 'prefix' part is no longer true :). maybe rename it to live_search_look_for(), live_search_look_for_substr() or somesuch.

also, the above change should not require 2 commits, please squash into one and force-push.

@MerlijnWajer - do you think we want this functionality?

freemangordon commented:
Oh, and IIUC, there is no need of tokenization, no? a simple strstr() on name/description should do the job, right?

TBH I am not sure I want that, but lets wait for others to comment as well.

le0nklcpp commented:
Thanks, changed function name. Yes, tokenization is not needed anymore. Also, there's some advanced search dialog in search.cc that is... not used?

le0nklcpp commented:
Tokenization removed here, if you decide to keep these changes, I can squash it too.

freemangordon commented:
Could you provide a usecase(package) for that change

le0nklcpp commented:
I'm not sure on what I'm asked as these changes only affect the small livesearch part. Replaced live_search_look_for with strstr, removed variables that were used to store tokenized values, in short it's just code optimization.

freemangordon commented:
Sorry for not being clear. My question was like - "could you provide example in which new behavior helps the user to find the package she looks for easier than without the change". I.e. - an example of new behavior improving the user experience.

For sure this is not code optimization, but functional change (looking for prefix vs looking for sub-string), unless I am missing something.

le0nklcpp commented:
There's the use case.
If you type "photo" you will find only "photo lightmeter" package,
with substring search HAM also finds "mihphoto" that is actually more relevant. There's not so many packages, so only few of these cases occur, and the search itself is not needed, but it's the default search dialog behavior in any package manager.
First commit will be enough for it. The second is only optimization.

freemangordon commented:
sorry, but I don't see that as a valid use case:
If mihphoto is https://github.com/halftux/MihPhoto/blob/master/qtc_packaging/debian_fremantle/control#L12 then you should search for "image" or "view", because it is image viewer and/or ask the maintainer to fix package description if you think it is wrong.

In regards to other package managers - just tested on my Ubuntu, Software Center finds nothing when search is for 'anoram' and a pile of packages when search is for 'panoram'

On the contrary, synaptic finds same packages for both 'anoram' and 'panoram'

However, I don't feel it right if I search for 'ale' to get 'morale', 'female' and 'calendar' as results :)

@MerlijnWajer: any thoughts?

le0nklcpp commented:
Another thing is livesearch was meant to be prefix search, so may be it is worth to add substring search instead of replacing. Even more, there is already a substring search that is not used. Exploring this right now

le0nklcpp committed:
Enable search dialog

le0nklcpp commented:
Without much problems search dialog can be re-enabled, so I'm closing this PR and leaving livesearch alone
